packages/datatrak-web-server/src/routes/SaveTaskRoute.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+import { Request } from 'express';
+import { KeysToCamelCase, Task } from '@tupaia/types';
+import { Route } from '@tupaia/server-boilerplate';
+
+export type SaveTaskRequest = Request<
+  Record<string, never>,
+  {
+    id: string;
+    message: string;
+  },
+  KeysToCamelCase<Task>,
+  Record<string, never>
+>;
+
+export class SaveTaskRoute extends Route<SaveTaskRequest> {
+  public async buildResponse() {
+    const { status } = this.req.body;
+    const { taskId } = this.req.params;
+    const { task } = this.req.models;
+
+    // Update task if id exists
+    if (taskId) {
+      await task.updateById(taskId, { status });
```

Comment:
Note: need to update to integrate with this [PR](https://github.com/beyondessential/tupaia/pull/5763/files#diff-cc52ec21ed7d955ee22da2eb18bf330290a5c79cf4f09d30f27ca17ae88c5151). Should also use the api class from context to ensure that it does through permission handling.